detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/CliArgument.kt

```diff
@@ -23,6 +24,13 @@ internal object CreateBaselineArgument : CliArgument() {
 	override fun toArgument() = listOf(CREATE_BASELINE_PARAMETER)
 }
 
+internal data class ClasspathArgument(val fileCollection: FileCollection) : CliArgument() {
+	override fun toArgument(): List<String> {
+		if (fileCollection.isEmpty()) return emptyList()
+		return listOf(CLASSPATH_PARAMETER, fileCollection.joinToString(",") { it.absolutePath })	
```

Comment:
I'm wary of the comma as path separator - to save cross-platform issues in future detekt should really follow the Java standard for path separators.

Safest would be to use `java.io.File.pathSeparatorChar` which is designed for this purpose. [The value is set dynamically](https://docs.oracle.com/javase/9/docs/api/java/io/File.html#pathSeparatorChar) based on platform (to `:` on *nix and `;` on Windows).
```suggestion
		return listOf(CLASSPATH_PARAMETER, fileCollection.joinToString(java.io.File.pathSeparatorChar) { it.absolutePath })	
```

Reply:
Sounds good to me, however would't this require double documentation/examples as the user must enter the right separator?

Reply:
This should be transparent to the user who will either not have to specify a classpath (the plugin should be able to figure it out on its own) and if they do specify manually it would be as a file collection, so no path separator necessary.

Even though it's internal though using the appropriate path separators should minimise ongoing issues with cross-platform support.